master/internal/task/allocation.go

```diff
@@ -241,7 +241,10 @@ func (a *Allocation) Receive(ctx *actor.Context) error {
 			a.Error(ctx, err)
 		}
 	case sproto.AllocationSignal:
-		a.HandleSignal(ctx, sproto.AllocationSignalWithReason{AllocationSignal: msg})
+		a.HandleSignal(ctx, sproto.AllocationSignalWithReason{
```

Comment:
Some context on ```sproto.AllocationSignal``` verse ```sproto.AllocationSignalWithReason```

It used to be just ```sproto.AllocationSignal``` which has this a vague "TERMINATED" or whatever message and with this we would try and do some guess work on the allocation state to add log message to try and explain what happened 

Then to eliminate this guess work a long time ago I added ```AllocationSignalWithReason``` to allow whoever is sending the signal to explain why they are sending the signal since they would have the most context on why.

This ```case sproto.AllocationSignal:``` was only left in in case I missed updating a spot to send a reason so we would send a blank reason rather than erroring the allocation (spoiler this is one spot I missed).

It would be great if this message was more specific like ```inactivity for more than for 5s```. To do this I would look for where we are sending the signal and update it to have a reason.

Think it gets sent here https://github.com/determined-ai/determined/blob/93a6af5dc4361b817f216fe40fb3e7adda589d3e/master/internal/task/idle_timeout.go#L42

Reply:
Noted, fix is now in `determined/master/internal/task/idle_timeout.go` rather than `determined/master/internal/task/allocation.go`.